maven-scm-test/src/main/java/org/apache/maven/scm/ScmTckTestCase.java

```diff
@@ -112,14 +114,19 @@ protected List<String> getScmFileNames()
     public abstract void initRepo()
         throws Exception;
 
+    private static final Logger LOG = LoggerFactory.getLogger( ScmTckTestCase.class );
+
     @Before
     public void checkScmPresence()
     {
         String scmProviderCommand = getScmProviderCommand();
         if ( scmProviderCommand != null )
         {
+            LOG.warn( "Checking for required SCM command \"{}\" --> {}",
+                    scmProviderCommand, ScmTestCase.isSystemCmd( scmProviderCommand ) );
             assumeTrue( "Skipping tests because the required command " + scmProviderCommand + " is not available.",
                 ScmTestCase.isSystemCmd( scmProviderCommand ) );
+            LOG.warn( "Required SCM command \"{}\" exists.", scmProviderCommand );
```

Comment:
Why warn? Is this intermediate?

Reply:
I added this just now to figure out why it still fails on macOS.

Reply:
I have just now uninstalled hg from my local system and I'm now able to locally reproduce the problem.

Reply:
Found it. Multiple `@Before` and no ordering guarantee among these.